src/app/app.routes.ts

```diff
@@ -21,6 +23,7 @@ export const rootRouterConfig: Routes = [
       { path: 'userPayment', component: UserPaymentComponent, canActivate: [AuthGuard]  },
       { path: 'status', component: StatusComponent, canActivate: [AuthGuard],  resolve: { data: StatusResolver} },
```

Comment:
data key in resolve is bad idea, then in the class component you get data.data.  
Replace it with:
resolve: { status: StatusResolver }